api/scpca_portal/models/dataset.py

```diff
@@ -59,3 +61,87 @@ class Meta:
 
     def __str__(self):
         return f"Dataset {self.id}"
+
+    @staticmethod
+    def validate_data(json_obj: Dict) -> bool:
+        data_validator = DataValidator(json_obj)
+        return data_validator.is_valid
+
+
+class DataValidator:
+    def __init__(self, data: Dict[str, Dict]) -> None:
+        self.data: Dict[str, Dict[str, Any]] = data
+
+    @property
+    def is_valid(self) -> bool:
+        return all(self.validate_project(project_id) for project_id in self.data.keys())
+
+    @property
+    def valid_projects(self) -> List[str]:
+        return [project_id for project_id in self.data.keys() if self.validate_project(project_id)]
+
+    @property
+    def invalid_projects(self) -> List[str]:
+        return [
+            project_id for project_id in self.data.keys() if not self.validate_project(project_id)
+        ]
+
+    def validate_project(self, project_id: str) -> bool:
+        is_valid = self._validate_project_id(project_id)
+        is_valid &= (
+            self._validate_merge_single_cell(project_id)
+            if self.data.get(project_id).get("merge_single_cell")
+            else True
+        )
+        is_valid &= (
+            self._validate_includes_bulk(project_id)
+            if self.data.get(project_id).get("includes_bulk")
+            else True
+        )
+        is_valid &= (
+            self._validate_single_cell(project_id)
+            if self.data.get(project_id).get(Modalities.SINGLE_CELL)
+            else True
+        )
+        is_valid &= (
+            self._validate_spatial(project_id)
+            if self.data.get(project_id).get(Modalities.SPATIAL)
+            else True
+        )
+
+        return is_valid
```

Comment:
Instead of evaluating everything before exiting with True/False, let's rewrite this to return false on first failed validation check. Base case return should be `True`.

Reply:
Can we also move "merge_single_cell", "includes_bulk" and the modalities into a new Enum called `DatasetDataProjectAttrs`